README.md

```diff
@@ -38,23 +38,37 @@ repos, _, err := client.Repositories.ListByOrg("github", opt)
 
 The go-github library does not directly handle authentication.  Instead, when
 creating a new client, pass an `http.Client` that can handle authentication for
-you.  The easiest and recommended way to do this is using the [goauth2][]
+you.  The easiest and recommended way to do this is using the [oauth2][]
 library, but you can always use any other library that provides an
 `http.Client`.  If you have an OAuth2 access token (for example, a [personal
-API token][]), you can use it with the goauth2 using:
+API token][]), you can use it with oauth2 using:
 
 ```go
-t := &oauth.Transport{
-  Token: &oauth.Token{AccessToken: "... your access token ..."},
+// create struct for the token source
+type tokenSource struct {
+  token *oauth2.Token
 }
 
-client := github.NewClient(t.Client())
+// add Token() method to satisfy oauth2.TokenSource interface
+func (t *tokenSource) Token() (*oauth2.Token, error){
+  return t.token, nil
+}
+
+func main() {
+ts := tokenSource{
```

Comment:
I wasn't using a pointer receiver for the Token method in my code so I didn't catch this. Thanks :)